vendor/github.com/openshift/hypershift/api/hypershift/v1beta1/nodepool_conditions.go

```diff
@@ -64,6 +64,10 @@ const (
 
 	// NodePoolValidMachineTemplateConditionType signal that the machine template created by the node pool is valid
 	NodePoolValidMachineTemplateConditionType = "ValidMachineTemplate"
+
+	// NodePoolClusterNetworkCIDRConflictType signals if a NodePool's machine objects are colliding with the
+	// cluster network's CIRD range. This can indicate why some network functionality might be degraded.
+	NodePoolClusterNetworkCIDRConflictType = "ClusterNetworkCIDRConflict"
```

Comment:
Checking the other conditions looks like the are designed around what we want to have like "ready, valid" instead of what we want to check "conflict", what do you think about chaging from conflict to valid ? 
```
NodePoolValidClusterNetworkCIDRValidType = "ValidClusterNetworkCIDR"
```

And we can talk about Conlict on the Reason and Message.

Reply:
I'm torn on that. My issue with this is that this condition is being posted to the NodePool and not the HostedCluster.

The NodePool isn't in charge of the ClusterNetwork cidr, so I found it a bit odd to say something isn't valid on an object that isn't in control of the value.

If i instead put this condition on the HostedCluster, I'd then need to make the HostedCluster controller start watching objects and triggering reconcile loops for objects (nodepools and machines) that the controller isn't acting on today. I don't feel great about that either.

my compromise was to report the cidr collision on the nodepool as more of an "informational" condition that isn't indicating a tuning is valid or not.

Reply:
Fair enough.